src/Functions/URL/decodeURLComponent.cpp

```diff
@@ -83,8 +108,8 @@ static size_t decodeURL(const char * src, size_t src_size, char * dst, bool plus
 
 
 /// Percent decode of URL data.
-template <bool plus_as_space>
-struct DecodeURLComponentImpl
+template <bool plus_as_space, bool encode>
```

Comment:
Please consider to use some strategy enum like
```
enum URLCodeStrategy
{
    encode,
    decode
}
```
and put `plus_as_space` as second template argument, because `strategy` is more important.

Reply:
done. got it!